### PR TITLE
Backport HA iotawatt integration changes

### PR DESCRIPTION
**Pull request by jyavenard**
This pull request is to be put on top of https://github.com/jyavenard/iotawatt_ha/tree/IntegrateLocally

It backports all the changes made into HA's official iotawatt integration and add the high resolution energy sensors.

### CODE REVIEW

**Comment by wishie** [comment]
So I have just updated HA to 2021.9.0 and am now using the 'built in' integration, but I have issues with sensors in the energy dashboard. The instructions on the HA page are not clear, or not working as expected. Can you explain what I need to do to have valid sensors for the energy dashboard?

**Comment by jyavenard** [comment]
If you are using a net solar system and need import and export ; the current version in iotawatt can't work.
The one in this branch can. 

Simply put the custom_components/iotawatt folder in the HA custom_components and restart HA

**Comment by wishie** [comment]
So Im not installing the 0.0.8 release, but a specific branch? How do I achieve this?

**Comment by gtdiehl** [comment]
> So Im not installing the 0.0.8 release, but a specific branch? How do I achieve this?

If you want to get up and running right now update the configuration with the integration sensor for each of your sensors from the IoTaWatt that has a Unit of ```Watts```

This is the example for the Home Assistant [webpage](https://www.home-assistant.io/integrations/iotawatt/)
```
sensor iotawatt:
  - platform: integration
    source: sensor.mainsexport
    name: Total Grid Export
    unit_prefix: k
  - platform: integration
    source: sensor.mainsconsumption
    name: Total Grid Consumption
    unit_prefix: k
```

In the above example there are two sensors on the IoTaWatt, ```sensor.mainsexport``` and ```sensor.mainsconsumption```. Replace those with the sensors from your IoTaWatt. Of course you probably have more than two, so just copy and paste and create as many given each one a unique name. Restart HA and the sensors with the names you gave will be available in the selection list for the Energy Dashboard

File: custom_components/iotawatt/sensor.py [hunk 1]
```diff
@@ -1,137 +1,266 @@
...
+        "Watts",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_POWER,
+    ),
+    "WattHours": IotaWattSensorEntityDescription(
+        "WattHours",
+        native_unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
```
Comment (gtdiehl) [inline custom_components/iotawatt/sensor.py:87]:
```suggestion
    "WattHours": IotaWattSensorEntityDescription(
        "WattHours",
        native_unit_of_measurement=ENERGY_WATT_HOUR,
        state_class=STATE_CLASS_TOTAL_INCREASING,
        device_class=DEVICE_CLASS_ENERGY,
    ),
```

File: custom_components/iotawatt/sensor.py [hunk 2]
```diff
@@ -1,137 +1,266 @@
...
-        return attrs
+        self._key = key
+        data = self._sensor_data
+        self._accumulating = data.getUnit() == "WattHours" and not data.getFromStart()
+        self._accumulated_value = None
+        if data.getType() == "Input":
+            unit = data.getUnit() + self._name_suffix
+            self._attr_unique_id = (
+                f"{data.hub_mac_address}-input-{data.getChannel()}-{unit}"
+            )
```
Comment (gtdiehl) [inline custom_components/iotawatt/sensor.py:176]:
```suggestion
        if data.getType() == "Input":
            unit = data.getUnit() + self._name_suffix
            self._attr_unique_id = (
                f"{data.hub_mac_address}-input-{data.getChannel()}-{unit}"
            )
        elif data.getType() == "Output":
            self._attr_unique_id = (
                f"{data.hub_mac_address}-output-{data.getName()}-{data.getUnit()}"
            ) 
```

File: custom_components/iotawatt/sensor.py (hunk 1 above)
Comment (jyavenard) [reply custom_components/iotawatt/sensor.py:87]: The class is already set on the high resolution sensor.

File: custom_components/iotawatt/sensor.py (hunk 2 above)
Comment (jyavenard) [reply custom_components/iotawatt/sensor.py:176]: Doing that would break transitioning to HA iotawatt integration though.

Comment (gtdiehl) [reply custom_components/iotawatt/sensor.py:176]: ok don't add it

**Comment by jyavenard** [comment]
this will only work with 2021.9 ; will approach this differently.

**Pull request closed without merging**